bot/exts/fun/adventure.py

```diff
@@ -0,0 +1,409 @@
+        await session.prepare()
+
+        return session
+
+    async def stop(self) -> None:
+        """Stops the game session, clean up by removing event listeners."""
+        self.cancel_timeout()
+        self._bot.remove_listener(self.on_reaction_add)
+        self._bot.remove_listener(self.on_message_delete)
+
+    @property
+    def is_in_ending_room(self) -> bool:
+        """Check if the game has ended."""
+        return self.current_room_data.get("type") == "end"
+
+    @property
+    def all_options(self) -> list[OptionData]:
+        """Get all options in the current room."""
+        return self.current_room_data.get("options", [])
+
+    @property
+    def available_options(self) -> bool:
+        """
+        Get "available" options in the current room.
+
+        This filters out options that require an effect that the user doesn't have or options that restrict an effect.
+        """
+        filtered_options = filter(
+            lambda option: (
+                "requires_effect" not in option or option.get("requires_effect") in self._effects
+            ) and (
+                "effect_restricts" not in option or option.get("effect_restricts") not in self._effects
+            ),
+            self.all_options
+        )
+
+        return filtered_options
+
+    @property
+    def current_room_data(self) -> RoomData | EndRoomData:
+        """Get the current room data."""
+        current_room = self._current_room
+
+        if current_room == "start":
+            return self.game_data[current_room]
+
+        return self.game_data["other_rooms"][current_room]
+
+    async def pick_option(self, index: int) -> None:
+        """Event that is called when the user picks an option."""
+        chosen_option = self.all_options[index]
+
+        next_room = chosen_option["leads_to"]
+        new_effect = chosen_option.get("effect")
+
+        # update all the game states
+        if new_effect:
+            self._effects.append(new_effect)
+        self._path.append(next_room)
+        self._current_room = next_room
+
+        # update the message with the new room
+        await self.update_message()
+
+
+class Adventure(DiscordCog):
+    """Custom Embed for Adventure RPG games."""
+
+    @commands.command(name="adventure")
+    async def new_adventure(self, ctx: Context, game_code_or_index: str | None = None) -> None:
+        """Wanted to slay a dragon? Embark on an exciting journey through text-based RPG adventure."""
+        try:
+            # prevent malicious pings and mentions
+            santiser = clean_content(fix_channel_mentions=True)
+            sanitised_game_code_or_index = await santiser.convert(ctx, game_code_or_index)
+
+            # quality of life: if the user accidentally wraps the game code in backticks, process it anyway
+            sanitised_game_code_or_index = sanitised_game_code_or_index.strip("`")
+
+            await GameSession.start(ctx, sanitised_game_code_or_index)
```

Comment:
Fixes a regression of `.adventure` not working because it is trying to sanitize a `None` object 😅
```suggestion
        if isinstance(game_code_or_index, str):
            # prevent malicious pings and mentions
            santiser = clean_content(fix_channel_mentions=True)
            game_code_or_index = await santiser.convert(ctx, game_code_or_index)

            # quality of life: if the user accidentally wraps the game code in backticks, process it anyway
            game_code_or_index = game_code_or_index.strip("`")
        try:
            await GameSession.start(ctx, game_code_or_index)
```